theme/management/commands/report_quota_inconsistency.py

```diff
@@ -0,0 +1,49 @@
+import csv
+import math
+from django.core.management.base import BaseCommand
+from django.core.exceptions import ValidationError
+from theme.models import UserQuota
+from hs_core.hydroshare.resource import get_quota_usage_from_irods
+
+
+class Command(BaseCommand):
+    help = "Output potential quota inconsistencies between iRODS and Django for all users in HydroShare"
+
+    def add_arguments(self, parser):
+        parser.add_argument('output_file_name_with_path', help='output file name with path')
+
+    def handle(self, *args, **options):
+        quota_report_list = []
+        for uq in UserQuota.objects.filter(
+                user__is_active=True).filter(user__is_superuser=False):
+            try:
+                used_value = get_quota_usage_from_irods(uq.user.username)
+                if not math.isclose(used_value, uq.used_value, abs_tol=0.1):
+                    # report inconsistency
+                    report_dict = {
+                        'user': uq.user.username,
+                        'django:': uq.used_value,
+                        'irods': used_value}
+                    quota_report_list.append(report_dict)
+                    print('quota incosistency: {} reported in django vs {} reported in iRODS for user {}'.format(
+                        uq.used_value, used_value, uq.user.username), flush=True)
+            except ValidationError as ex:
+                print(ex, flush=True)
+
+        if quota_report_list:
```

Comment:
What do you think about making the output file argument optional and only writing to console if it isn't provided?

Reply:
@sblack-usu I thought about this initially, and convinced myself that it'd always be a good thing to write the inconsistency to a file in addition to the console if there were any inconsistencies. This will guarantee the inconsistencies are captured in a file and prevent the potential loss of inconsistency info if for some reason console outputs were not captured in time.